src/MapboxStyleParser.ts

```diff
@@ -699,11 +742,39 @@ export class MapboxStyleParser implements StyleParser<Omit<MbStyle, 'sources'>>
           scaleDenominator,
           symbolizers
         };
-        gsRules?.push(rule);
+        gsRules.push(rule);
+        if (layer.source && typeof layer.source === 'string') {
+          if (!sourceMapping[layer.source]) {
+            sourceMapping[layer.source] = [];
+          }
+          sourceMapping[layer.source].push(gsRules.length - 1);
+        }
+        if (layer['source-layer']) {
+          if (!sourceLayerMapping[layer['source-layer']]) {
+            sourceLayerMapping[layer['source-layer']] = [];
+          }
+          sourceLayerMapping[layer['source-layer']].push(gsRules.length - 1);
+        }
+        if (symbolizers.length > 1) {
+          splitSymbolizers.push({
+            rule: gsRules.length - 1,
+            symbolizers: symbolizers.map((s, sIdx) => sIdx)
+          });
+        }
       });
     }
 
-    return gsRules || [];
+    if (splitSymbolizers.length) {
```

Comment:
Function arguments should not be mutated. Return a clone with the new values instead.